net/nginx-util/src/px5g.cpp

```diff
@@ -0,0 +1,366 @@
+#include <string>
+#include <iostream>
+#include <unistd.h>
+using namespace std;
+#include "px5g-openssl.hpp"
+
+
+int checkend(const char * argv[]);
+
+
+void eckey(const char * argv[]);
+
+
+void rsakey(const char * argv[]);
+
+
+void selfsigned(const char * argv[]);
+
+
+inline int parse_int(string arg)
+{
+    size_t pos;
+    int ret = stoi(arg, &pos);
+    if (pos < arg.size()) {
+        throw runtime_error("number has trailing char");
+    }
+    return ret;
+}
+
+
+inline auto parse_curve(string name)
+{
+    if (name=="P-256" || name=="secp256r1") { return NID_X9_62_prime256v1; }
+    else if (name=="secp192r1") { return NID_X9_62_prime192v1; }
```

Comment:
else if after return is pointless. This look like it can be replaced by a simple switch statement.

Reply:
Fixed that :-)